version/version.go

```diff
@@ -0,0 +1,23 @@
+/*
+Copyright (c) 2014-2022 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+// SDKName is the name of this SDK
+const SDKName = "govmomi"
+
+// SDKVersion is the version of this SDK
+const SDKVersion = "0.28.0"
```

Comment:
Maybe `Client` instead of `SDK` ?
```go
const ClientName = "govmomi"
const ClientVersion = "0.28.0"
const UserAgent = ClientName + "/" + ClientVersion
```

Reply:
Thank you this is great step.

I wonder if there should be a part that is easy to customize for client to set name and version distinct from govmomi. Also a part that is platform specific. It is rare that platform like golang will cause issues or OS compatibility is an issue. Yet I have seen some issues that show up with dual protocol IPv4/IPv6 setup on Windows as it behaves subtly different from Linux in corner case scenarios.

I can imagine the user-agent ends up looking something like the following to help in support scenarios 

`Terraform vSphere Provider/2.1.1 govmomi/0.28.0 golang/1.18.3 (MacOS; Darwin Kernel Version 21.5.0; x86_64)`

I can volunteer time to figure out the part after govmomi.

I can also imagine clients may want to override the whole header as they can by setting UserAgent in order to not disclose implementation details.

Reply:
I like your ideas!

Regarding user-agent we can get some inspiration from https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent

Will respond later to other points.

Reply:
The following RFC defines the format https://datatracker.ietf.org/doc/html/rfc7231#section-5.5.3

As I understand it the format is a list of products separated with white space. Each product has optional version after  slash character and optional comments in brackets.

Reply:
I've settled on using the format `govmomi/0.28.0 (go1.18.3;linux;amd64)` (follows AWS SDK convention)

Reply:
> I can also imagine clients may want to override the whole header as they can by setting UserAgent in order to not disclose implementation details.

This is already supported by overwriting the default, i.e. no breaking change.